src/bun.js/api/bun/socket.zig

```diff
@@ -2581,14 +2627,18 @@ fn NewSocket(comptime ssl: bool) type {
         }
 
         pub fn jsRef(this: *This, globalObject: *JSC.JSGlobalObject, _: *JSC.CallFrame) bun.JSError!JSValue {
+            log("ref {s}", .{if (this.handlers.is_server) "S" else "C"});
             JSC.markBinding(@src());
+            if (this.socket.isDetached()) this.poll_ref_before_connect += 1;
             if (this.socket.isDetached()) return JSValue.jsUndefined();
             this.poll_ref.ref(globalObject.bunVM());
             return JSValue.jsUndefined();
         }
 
         pub fn jsUnref(this: *This, globalObject: *JSC.JSGlobalObject, _: *JSC.CallFrame) bun.JSError!JSValue {
+            log("unref {s}", .{if (this.handlers.is_server) "S" else "C"});
             JSC.markBinding(@src());
+            if (this.socket.isDetached()) this.poll_ref_before_connect -= 1;
```

Comment:
This should definitely not be a count. It's not a count in general, it's an enum. I think `poll_ref_before_connect` should be removed

Reply:
fixing and adding a test